### ISSUE TITLE
False positive for `await_holding_refcell_ref`

### ISSUE THREAD

**Issue by Daniel-B-Smith** [rust-lang/rust-clippy#6353]
I tried this code:

```rust
let mut state = self.state.borrow_mut();
let handles = state.cancel_all_in_flight();
drop(state);
let to_cancel = handles.len();
let cancelled = stream::iter(handles).then(|f| f).count().await;
```

I expected to see this happen: No error since the `RefMut` is explicitly dropped before the `await`

Instead, this happened: Brought up a warning

This happens in the recently promoted beta.

This should be pretty easy to fix. As a short term mitigation, I will downgrade the check from correctness and should hopefully be able to work on the fix next week.

**Comment by khuey** [comment]
Any progress here?

**Comment by Daniel-B-Smith** [comment]
Apologies for the lack of communication. From conversations on Zulip, it is in fact not an easy fix to identify variables that have been explicitly dropped. I intended to add the documentation quickly, but got caught up in things. I just sent out https://github.com/rust-lang/rust-clippy/pull/6585. Unfortunately, I don't have the cycles to dig into this deeply right now.

**Comment by Ralith** [comment]
This lint seems to have been enabled by default in 1.61 and is still producing false positives, breaking my CI.

**Comment by Ralith** [comment]
Bump. This is still breaking my CI. Can we get it rolled back?

**Comment by cbiffle** [comment]
I believe this class of false positive was fixed -- I'm working on code that's doing explicit dropping of RefMuts in complex loops without any false positives from Clippy (though I have gotten a couple of true-positives!).

**Comment by AaronDewes** [comment]
I've still been getting such false positives with Rust 1.91 ( `clippy 0.1.91 (f8297e351a 2025-10-28)` ), so it does not appear to be fully fixed.

**Comment by Jarcho** [comment]
This still has no fix. If the scope of a binding extends past an await point it's erroneously assumed to be live by this lint. This isn't trivial to fix since this requires dataflow analysis to determine the value of any drop flags at await points. Alternatively the compiler could be changed to not store such values in the generator, although I'm not sure how that interacts with potential panics.
